File: src/infrastructure/userIdentitiesRepository.js
var userIdentity = require('../domain/identity/userIdentity');

var UnknownUserIdentity = exports.UnknownUserIdentity = function UnknownUserIdentity(userId){
    this.userId = userId;
};

var MessageRepository = function MessagesRepository(eventsStore){
    var getAllEvents = function getAllEvents(userId){
        var events = eventsStore.getEventsOfAggregate(userId);
        if(!events.length){
            throw new UnknownUserIdentity(userId);
        }

        return events;
    };

    this.getUserIdentity = function getUserIdentity(userId){
        var events = getAllEvents(userId);
        return userIdentity.create(events);
    };
};

exports.create = function create(eventsStore){
    return new MessageRepository(eventsStore);
};

File: Mixter.Infrastructure/UserIdentitiesRepository.cs
using System.Linq;
using Mixter.Domain.Identity;

namespace Mixter.Infrastructure
{
    public class UserIdentitiesRepository : IUserIdentitiesRepository
    {
        private readonly EventsStore _eventStore;

        public UserIdentitiesRepository(EventsStore eventStore)
        {
            _eventStore = eventStore;
        }

        public UserIdentity GetUserIdentity(UserId userId)
        {
            var events = _eventStore.GetEventsOfAggregate(userId).ToArray();
            if (!events.Any())
            {
                throw new UnknownUserIdentity(userId);
            }

            return new UserIdentity(events);
        }
    }
}
